CustomLogListener.cs

```diff
@@ -0,0 +1,145 @@
+using BepInEx.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace RainMeadow
+{
+    public partial class RainMeadow
+    {
+        public class CustomLogListener : ILogListener
+        {
+            public static HashSet<string> BlacklistedSources = new() { "Unity", "Unity Log" };
+
+            public CustomLogListener(string localPath, int maxLogFiles = 2, string? rootPath = null, bool delayedFlushing = false)
+            {
+                rootPath ??= BepInEx.Paths.BepInExRootPath;
+
+                try
+                {
+                    var logFiles = Directory.GetFiles(rootPath, "meadowLog.*.log");  // assume sorted from oldest to newest
+                    if (logFiles.Count() >= maxLogFiles)
+                    {
+                        foreach (var path in logFiles.Take(logFiles.Count() - maxLogFiles + 1))
+                        {
+                            File.Delete(path);
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    Error($"couldn't clean up meadowLog files: {e}");
+                }
+
+                DisplayedLogLevel = LogLevel.Info;
+
+                if (!BepInEx.Utility.TryOpenFileStream(Path.Combine(rootPath, localPath),
+                    FileMode.Create, out var fileStream, share: FileShare.Read, access: FileAccess.Write))
+                {
+                    Error($"couldn't open logfile {localPath}!");
+                    return;
+                }
+
+                LogWriter = TextWriter.Synchronized(new StreamWriter(fileStream, BepInEx.Utility.UTF8NoBom));
+
+                if (delayedFlushing) FlushTimer = new Timer(o => { LogWriter?.Flush(); }, null, 2000, 2000);
+
+                InstantFlushing = !delayedFlushing;
+            }
+
+            public LogLevel DisplayedLogLevel { get; }
+
+            public TextWriter LogWriter { get; protected set; }
+
+            private Timer FlushTimer { get; }
+
+            private bool InstantFlushing { get; }
+
+            public LogLevel LogLevelFilter => DisplayedLogLevel;
+
+            public string Strip(string line)
+            {
+                // strip timestamps
+                if (line.Substring(8, 13) == ":RainMeadow] " && line[29] == '|' && line[36] == '|')
```

Comment:
I don't think these indexes are fixed, with how the tick number grows in digits.

It might be simpler to move the timestamp stuff to the custom logger

Reply:
whoops yeah, the second '|' isn't fixed

Reply:
if we move timestamps to our custom logger then it won't appear in the normal LogOutput, are we ok with that?